src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/Service/NavigationService.show.cs

```diff
@@ -203,35 +214,89 @@ protected async Task<TViewModel> InitializeViewModelAsync<TViewModel>()
 
 
 
+        private Func<Page, Task> BuildShowAsyncFunc(
+            Page page)
+        {
+            if (RootPage is MasterDetailPage &&
+                page is DetailPage)
+            {
+                return SetDetailAsync;
+            }
+            else if (RootPage is TabbedPage &&
+                page is TabPage)
+            {
+                return SetTabAsync;
+            }
+            else return PushAsync;
+        }
 
-        internal static async Task<TViewModel> InitializeViewModelInternal<TViewModel>()
 
-            where TViewModel : class, IViewModel
+        private async Task PushAsync(
+            Page page)
         {
-            var viewModelInstance = dependencyResolver.Resolve<TViewModel>();
-            await viewModelInstance.InitializeAsync();
+            if (Navigation == null)
+                return;
 
-            Log?.Info(
-                $"ViewModel viewModel of type {typeof(TViewModel).Name} has been created and initialized!");
+            await Device.InvokeOnMainThreadAsync(() =>
+            {
+                Navigation.PushAsync(
+                    page,
+                    animated: Configuration.UseAnimations);
+            });
+        }
 
-            return viewModelInstance;
+        private async Task SetDetailAsync(
+            Page page)
+        {
+            if (!(Application.Current.MainPage is MasterDetailPage masterDetailPage))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)